File: euler010.py
#!/usr/bin/python
import math

def eratosthenes(cap):
    a = []
    for i in xrange(0, cap + 1):
        a.append(i % 2 == 1)
    for i in xrange(3, int(math.ceil(math.sqrt(cap + 1))), 2):
        if a[i] == True:
            for j in xrange(i * i, cap + 1, i):
                a[j] = False
    primes = []
    for i in xrange(2, len(a)):
        if a[i]:
            primes.append(i)
    return primes

primes = eratosthenes(2000000)
print sum(primes)

File: euler010.go
package main

import (
	"fmt"
	"math"
)

func eratosthenes(cap int) []int {
	A := make([]bool, cap)
	for i := 0; i < cap; i++ {
		A[i] = true
	}

	for i := 2; i < int(math.Sqrt(float64(cap))); i++ {
		if A[i] {
			for j := i * i; j < cap; j += i {
				A[j] = false
			}
		}
	}

	result := make([]int, 0)
	for i, val := range A {
		if i >= 2 && val {
			result = append(result, i)
		}
	}
	return result
}

func main() {
	sum := 0
	for _, val := range eratosthenes(2000000) {
		sum += val
	}
	fmt.Println(sum)
}
